extension/jmxmetricextension/integration_test.go

```diff
@@ -0,0 +1,358 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package jmxmetricextension
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"runtime"
+	"testing"
+	"time"
+
+	prommodel "github.com/prometheus/common/model"
+	promconfig "github.com/prometheus/prometheus/config"
+	promdiscovery "github.com/prometheus/prometheus/discovery"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.opentelemetry.io/collector/receiver/prometheusreceiver"
+	"go.opentelemetry.io/collector/testbed/testbed"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+type JmxIntegrationSuite struct {
+	suite.Suite
+	JarPath string
+}
+
+func TestJmxIntegration(t *testing.T) {
+	suite.Run(t, new(JmxIntegrationSuite))
+}
+
+func (suite *JmxIntegrationSuite) SetupSuite() {
+	jarPath, err := downloadJmxMetricGathererJar()
+	require.NoError(suite.T(), err)
+	suite.JarPath = jarPath
+}
+
+func (suite *JmxIntegrationSuite) TearDownSuite() {
+	require.NoError(suite.T(), os.Remove(suite.JarPath))
+}
+
+func downloadJmxMetricGathererJar() (string, error) {
+	url := "https://oss.jfrog.org/artifactory/list/oss-snapshot-local/io/opentelemetry/contrib/opentelemetry-java-contrib-jmx-metrics/0.0.1-SNAPSHOT/opentelemetry-java-contrib-jmx-metrics-0.0.1-20200918.184353-3.jar"
```

Comment:
How big is this JAR? Is it feasible to store it in this repo, or download it as part of the build? In any case, would be good to have a retry logic ;-)

Reply:
Currently it's about 20MB: https://oss.jfrog.org/artifactory/oss-snapshot-local/io/opentelemetry/contrib/opentelemetry-java-contrib-jmx-metrics/0.0.1-SNAPSHOT/

Bundling/packaging is being evaluated and would be introduced in a later PR.  It's my impression that git is not a good way to store binaries so downloading (and potentially caching in CI) is likely going to be necessary overall.